### PR TITLE
[OS-2725] adding ctx to camunda subscriptions

### PR DESCRIPTION
**Pull request by shruthiMaganahalli**

### CODE REVIEW

**Comment by shane-blacklane** [comment]
Code changes look good, but you need to follow the PR standards on this repo: https://github.com/blacklane/go-libs#commit-messages-issues-and-pull-requests

File: camunda/client.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func (c *client) SendMessage(ctx context.Context, messageType string, businessKe
 	return nil
 }
 
-func (c *client) Subscribe(topicName string, workerID string, handler TaskHandler, options ...func(*Subscription)) *Subscription {
+func (c *client) Subscribe(ctx context.Context, topicName string, workerID string, handler TaskHandler, options ...func(*Subscription)) *Subscription {
```
Comment (stephde) [inline camunda/client.go:80]: I'm not sure if it makes sense to pass a context to this function. This is only called once when the service starts and initializes the subscription. Therefore, any `TaskHandlerFunc` triggered from this function will receive the same context instance.
I'm not sure if this is a problem, but I feel like it makes more sense to have separate contexts for each handler call. So IMHO it makes more sense to generate a new context in the `fetch()` function. The events lib is handling this in the same way so I think we should stick to that
https://github.com/blacklane/go-libs/blob/master/x/events/consumer.go#L201
One main thing we want to achieve with the context is to pass the logger indirectly. So we would have to pass the logger instance here instead and manually inject that into the context. But I suppose the library should not really know about the logger. The events library solves this by introducing middlewares. I guess that would be the best way to solve this here as well. However thats a bit more effort. Let me know your thoughts on this 
🥔

**Review by stephde — CHANGES_REQUESTED** [review]

File: camunda/client.go (hunk 1 above)
Comment (sabaraouf) [reply camunda/client.go:80]: i dont see the difference in the 2 PRs https://github.com/blacklane/go-libs/pull/97. also stephan's suggestion was to create a new context in `fetch()` but it doesn't look like so 🤔

**Comment by shane-blacklane** [comment]
closing due to inactivity

**Pull request closed without merging**